File: frontend/src/types/skill.ts
// 技能模块类型定义

// 问答示例
export interface SkillExample {
  q: string  // 问题
  a: string  // 答案
}

// 关联脚本
export interface SkillScript {
  name: string    // 脚本名称
  content: string // 脚本代码内容
}

// 技能
export interface Skill {
  id: string
  ico: string                    // 图标
  name: string                   // 技能名称
  scope: 'builtin' | 'custom'    // 范围（内置/自定义）
  ver: string                    // 版本号
  desc: string                   // 描述
  trigger: string                // 触发条件
  prompt: string                 // 系统 Prompt
  tools: string[]                // 挂载的工具 ID 列表
  docs: string[]                 // 挂载的文档 ID 列表
  wfs: string[]                  // 挂载的工作流 ID 列表
  examples: SkillExample[]       // 问答示例
  scripts: SkillScript[]         // 关联脚本
  budget?: number                // Token 预算（可选）
  used?: number                  // 已用 Token（可选）
}

File: backend/app/schemas/skill.py
"""技能模块 Pydantic schema（对齐前端 types/skill.ts）。"""
from pydantic import BaseModel


class SkillExample(BaseModel):
    q: str
    a: str


class SkillScript(BaseModel):
    name: str
    content: str


class SkillCreate(BaseModel):
    ico: str = "🔧"
    name: str
    scope: str = "custom"
    ver: str = "1.0.0"
    desc: str = ""
    trigger: str = ""
    prompt: str = ""
    tools: list[str] = []
    docs: list[str] = []
    wfs: list[str] = []
    examples: list[SkillExample] = []
    scripts: list[SkillScript] = []
    budget: int | None = None


class SkillUpdate(BaseModel):
    ico: str | None = None
    name: str | None = None
    scope: str | None = None
    ver: str | None = None
    desc: str | None = None
    trigger: str | None = None
    prompt: str | None = None
    tools: list[str] | None = None
    docs: list[str] | None = None
    wfs: list[str] | None = None
    examples: list[SkillExample] | None = None
    scripts: list[SkillScript] | None = None
    budget: int | None = None


class SkillOut(BaseModel):
    id: str
    ico: str
    name: str
    scope: str
    ver: str
    desc: str
    trigger: str
    prompt: str
    tools: list[str]
    docs: list[str]
    wfs: list[str]
    examples: list[SkillExample]
    scripts: list[SkillScript]
    budget: int | None = None
    used: int | None = None
